12_graphs/graph.js

```diff
@@ -32,6 +32,41 @@ class Graph {
         }
       });
     }
+    return visitedArr;
+  };
+  
+  depthFirstTraversal(startVertex) {
+    if (!startVertex) { throw new Error('Error: no start vertex provided') }
+    if (!this.vertices[startVertex]) { throw new Error('Error: vertex not in graph') }
+
+    let traversalStack = [startVertex];
+    let visitedArr = [startVertex];
+    let currUnvisitedChild = false;
+    let _peak = (array) => {

```

Comment:
Bad practice (not clean code) to name variables with the same name as primitive types. So no `array`, `number`, `string`, etc. Also in this case, it'd be better to focus on the context for the name rather than the implementation.